src/phongo_bson_encode.c

```diff
@@ -141,7 +164,7 @@ static void php_phongo_bson_append_object(bson_t* bson, php_phongo_field_path* f
 
 			/* Persistable objects must always be serialized as BSON documents;
 			 * otherwise, infer based on bsonSerialize()'s return value. */
-			if (instanceof_function(Z_OBJCE_P(object), php_phongo_persistable_ce) || php_phongo_is_array_or_document(&obj_data) == IS_OBJECT) {
+			if (instanceof_function(Z_OBJCE_P(object), php_phongo_persistable_ce) || php_phongo_is_array_or_document(&obj_data) != IS_ARRAY) {
```

Comment:
I think it'd also help for `php_phongo_is_array_or_document` to assert that only an object or array is passed in. We don't really have to worry about this thanks to the explicit checks on `obj_data` above, but it'd be good for peace of mind. I'll make a note of this in [PHPC-2217](https://jira.mongodb.org/browse/PHPC-2217) as well.